Source/Dafny/Verifier/Translator.cs

```diff
@@ -11143,7 +11143,7 @@ public enum K { Free, Checked, Both }
 
       } else if (expr is UnaryOpExpr) {
         var e = (UnaryOpExpr)expr;
-        if (e.Op == UnaryOpExpr.Opcode.Not) {
+        if (e.ResolvedOp == UnaryOpExpr.ResolvedOpcode.BoolNot) {
```

Comment:
what about `BVNot`, wouldn't the previous code apply to that as well?

Reply:
No, because this function has a `Contracts.Requires(expr.Type.IsBoolType)`

Reply:
Got it.